h2o-logger/src/test/java/LoggerTest.java

```diff
@@ -0,0 +1,49 @@
+import water.logging.ConsoleLogger;
+import water.logging.ILogger;
+import water.logging.Logger;
+import water.logging.Slf4JLogger;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.Assert.assertTrue;
+
+public class LoggerTest {
+    
+    @Test
+    public void noSlf4JAvailable() {
+        ILogger logger = Logger.getLogger(LoggerTest.class);
+        assertTrue(logger instanceof ConsoleLogger);
+    }
+
+    @Test
+    public void Slf4JAvailable() throws NoSuchMethodException, InvocationTargetException, IllegalAccessException, IOException {
+        ClassLoader classLoader = LoggerTest.class.getClassLoader();
+        
+        URL apiUrl = classLoader.getResource("slf4j-api-1.7.29.jar");
```

Comment:
I would just do set JUnit tests instead of this one

if you had a LoggerFactory, you wouldn't need to have the Logger class all static

if it was not static you can parametrize the SLF4J class that is being checked (and you can use any dummy class) to get ConsoleLogger instead of manipulating the classpath

Reply:
Logger class parametrized